src/cogent3/core/new_sequence.py

```diff
@@ -1774,59 +1773,32 @@ def get_translation(
         ------
         AlphabetError if include_stop is False and a stop codon occurs
         """
-        # todo gavin the genetic code should have the moltype for protein
-        #  with stop. The genetic code should also do the translation off
-        #  raw data, either an array of ints or a string. So this method
-        #  should only deal with trimming terminal stops, modulo 3 etc...
+
         from cogent3.core import new_genetic_code, new_moltype
 
         protein = new_moltype.get_moltype(
             "protein_with_stop" if include_stop else "protein"
         )
         gc = new_genetic_code.get_code(gc)
-        codon_alphabet = gc.get_alphabet(include_stop=include_stop).with_gap_motif()
-        moltype = self.moltype
-        # translate the codons
-        translation = []
-        if include_stop or not trim_stop:
-            # we just deal with sequence as is
-            seq = str(self)
+
+        if trim_stop:
+            seq = self.trim_stop_codon(gc=gc, strict=not incomplete_ok)
         else:
-            seq = str(self.trim_stop_codon(gc=gc, strict=not incomplete_ok))
+            seq = self
 
-        for posn in range(0, len(seq) - 2, 3):
-            orig_codon = str(seq[posn : posn + 3])
-            try:
-                resolved = moltype.resolve_ambiguity(
-                    orig_codon, alphabet=codon_alphabet
-                )
-            except new_alphabet.AlphabetError:
-                if not incomplete_ok or "-" not in orig_codon:
-                    raise new_alphabet.AlphabetError(
-                        f"unresolvable codon {orig_codon!r} in {self.name}"
-                    )
-                resolved = (orig_codon,)
-            trans = []
-            for codon in resolved:
-                if codon == "---":
-                    aa = "-"
-                elif "-" in codon:
-                    aa = "?"
-                    if not incomplete_ok:
-                        raise new_alphabet.AlphabetError(
-                            f"incomplete codon {codon} in {self.name}"
-                        )
-                else:
-                    aa = gc[codon]
-                    if aa == "*" and not include_stop:
-                        continue
-                trans.append(aa)
-            if not trans:
-                raise new_alphabet.AlphabetError(orig_codon)
-            aa = protein.what_ambiguity(trans)
-            translation.append(aa)
-
-        return protein.make_seq(seq="".join(translation), name=self.name)
+        # since we are realising the string, reverse complementing will be
```

Comment:
The genetic code method type hint indicates it can be "StrORBytesORArray". In the interest of seeing whether this will work, try using `numpy.array(seq)` instead.

Reply:
This will work once `Sequence.__str__` complements the sequence